### PR TITLE
Added suggestion filter by prefix

### PR DESCRIPTION
**Pull request by lexmihaylov**
rsense is not always returning results, so filtering the already suggested list will do the trick.

### CODE REVIEW

**Comment by ianhattendorf** [comment]
Thanks, sorry for the late response. I've been swamped lately, hopefully I have some time to check it out this weekend.

**Comment by lexmihaylov** [comment]
Hey, I've been doing some more modifications and error handling. I'll shoot a pull request shortly. I'm also thinking of adding online documentation support.

**Comment by ianhattendorf** [comment]
Sounds good, I'm out of town until the 31st so I'll take a look at everything then.

File: lib/autocomplete-ruby-client.coffee [hunk 1]
```diff
@@ -28,6 +29,9 @@ class RsenseClient
       dataType: 'json'
       data: JSON.stringify request
       error: (jqXHR, textStatus, errorThrown) ->
+        # send empty array to callback
```
Comment (ianhattendorf) [inline lib/autocomplete-ruby-client.coffee]: Could you remove the trailing whitespace here?

File: lib/autocomplete-ruby-provider.coffee [hunk 2]
```diff
@@ -1,5 +1,8 @@
 RsenseClient = require './autocomplete-ruby-client.coffee'
 
+String.prototype.regExpEscape = () ->
+  return @replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```
Comment (ianhattendorf) [inline lib/autocomplete-ruby-provider.coffee]: Semicolons are invalid in coffeescript and need to be removed

File: lib/autocomplete-ruby-provider.coffee [hunk 3]
```diff
@@ -8,6 +11,7 @@ class RsenseProvider
 
   constructor: ->
     @rsenseClient = new RsenseClient()
+    @lastSuggestions = [];
```
Comment (ianhattendorf) [inline lib/autocomplete-ruby-provider.coffee]: Another semicolon

File: lib/autocomplete-ruby-provider.coffee [hunk 4]
```diff
@@ -17,21 +21,43 @@ class RsenseProvider
...
           prefix: prefix
           label: "#{kind} (#{completion.qualified_name})"
         suggestions.push(suggestion)
+
       return suggestions
     return []
 
+
+  filterSuggestions: (prefix, suggestions) ->
+    suggestionBuffer = [];
```
Comment (ianhattendorf) [inline lib/autocomplete-ruby-provider.coffee]: Semicolon

File: lib/autocomplete-ruby-provider.coffee [hunk 5]
```diff
@@ -17,21 +21,43 @@ class RsenseProvider
...
+
+    if(!prefix?.length || !suggestions?.length)
+      return []
+
+    expression = new RegExp("^"+prefix.regExpEscape(), "i");
```
Comment (ianhattendorf) [inline lib/autocomplete-ruby-provider.coffee]: Semicolon, also a space on each side of the + would be nice

**Comment by ianhattendorf** [comment]
Looks good besides the whitespace and semicolons. I tested it out and it seems like completion is working after '.' and '::' now. If you could update your pull request with the fixes I noted I'll go ahead and merge it in.

**Comment by lexmihaylov** [comment]
Done! ;)

**Pull request merged**